build.gradle.kts

```diff
@@ -69,5 +71,39 @@ subprojects {
                 junitXml.isOutputPerTestCase = true
             }
         }
+        "uploadArchives"(Upload::class) {
```

Comment:
upload archives is the deprecated publishing mechanism - any reason we can't use the new one? For that matter, most of this looks like it's just changing the destination used by our regular publish plugin - can we use (or modify that)?  This isn't the kind of stuff we want copied around repo to repo.

Reply:
I've followed the official documentation for this task: https://central.sonatype.org/pages/gradle.html

Reply:
That's horribly outdated. The upload mechanism was deprecated way back in gradle 1 in 2014 - its remained for legacy support (probably this documentation 🙄 . There's not too much to  using the new mechanism, I can help you out.

See example here: https://docs.gradle.org/current/userguide/publishing_maven.html#publishing_maven:complete_example